### PR TITLE
WLA-595: Adds Discount to Order

### PR DESCRIPTION
**Pull request by josippetric**
Needed to implement WLA-595
https://guestlogix.atlassian.net/browse/WLA-595

**Description**

**Issues that should be CLOSED by merge of this PR:**
* Fixes #

**Related issues that should be LINKED to this PR:**
* Connects #

### CODE REVIEW

**Review by rhoque-glx — APPROVED** [review]

File: traveler-swift-core/TravelerKit/Models/Discount.swift [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    /// The `Discount` is being processed
+    case pending
+    /// The `Discount` is applied and is visible in total price calculation
+    case applied
+    /// The `Discount` status is unknown
+    case unknown
+}
+
+/// Holds information about a discount.
+public struct Discount: Decodable {
```
Comment (OmarPadierna) [inline traveler-swift-core/TravelerKit/Models/Discount.swift]: It should be named DiscountCoupon

File: traveler-swift-core/TravelerKit/Models/Order.swift [hunk 2]
```diff
@@ -50,6 +50,8 @@ public struct Order: Decodable, Equatable, Hashable {
     public let contact: CustomerContact
     /// The additional description needed for some orders
     public var description: String?
+    /// The optional `Discount` for the Order
+    public let discount: Discount?
```
Comment (OmarPadierna) [inline traveler-swift-core/TravelerKit/Models/Order.swift]: public let coupon: DiscountCoupon

**Review by OmarPadierna — CHANGES_REQUESTED** [review]

File: traveler-swift-core/TravelerKit/Models/Discount.swift (hunk 1 above)
Comment (josippetric) [reply traveler-swift-core/TravelerKit/Models/Discount.swift]: Done

File: traveler-swift-core/TravelerKit/Models/Order.swift (hunk 2 above)
Comment (josippetric) [reply traveler-swift-core/TravelerKit/Models/Order.swift]: Done

**Comment by josippetric** [comment]
@OmarPadierna I've made the change. Discount is not DiscountCoupon.

**Review by OmarPadierna — APPROVED** [review]

**Pull request merged**